src/Compilers/CSharp/Portable/Symbols/ConstructedNamedTypeSymbol.cs

```diff
@@ -54,6 +60,11 @@ internal ConstructedNamedTypeSymbol(NamedTypeSymbol constructedFrom, ImmutableAr
             Debug.Assert(constructedFrom.Arity != 0);
         }
 
+        protected override NamedTypeSymbol WithTupleDataCore(TupleUncommonData newData)
+        {
+            return new ConstructedNamedTypeSymbol(_constructedFrom, _typeArgumentsWithAnnotations, IsUnboundGenericType) { _lazyTupleData = newData };
```

Comment:
>return new ConstructedNamedTypeSymbol(_constructedFrom, _typeArgumentsWithAnnotations, IsUnboundGenericType) { _lazyTupleData = newData } [](start = 12, length = 137)

Consider adding an optional `TupleUncommonData` argument to the constructors for the few types that can represent a tuple, and change `_lazyTupleData` to a `private readonly` field.